component/core/src/main/java/org/exoplatform/social/core/service/LinkProvider.java

```diff
@@ -588,4 +592,62 @@ public static String getResourceBundleLabel(Locale locale, String label) {
     ResourceBundleService resourceBundleService =  ExoContainerContext.getService(ResourceBundleService.class);
     return resourceBundleService.getResourceBundle(resourceBundleService.getSharedResourceBundleNames(), locale).getString(label);
   }
+
+  public static String getBaseURLSiteRest() {
+    return "/" + PortalContainer.getCurrentPortalContainerName() + "/" + PortalContainer.getCurrentRestContextName()
+        + BASE_URL_SITE_REST_API;
+  }
+
+  public static String buildSiteBannerUrl(String siteName, long bannerFileId) {
+    if (StringUtils.isBlank(siteName)) {
+      return null;
+    }
+    String token = generateSiteBannerToken(siteName, ATTACHMENT_BANNER_TYPE, bannerFileId);
+    if (StringUtils.isNotBlank(token)) {
+      try {
+        token = URLEncoder.encode(token, "UTF8");
+      } catch (UnsupportedEncodingException e) {
+        LOG.warn("Error encoding token", e);
+        token = StringUtils.EMPTY;
+      }
+    }
+    boolean isDefault = bannerFileId == 0 ;
+    return new StringBuilder(getBaseURLSiteRest()).append("/")
+                                                  .append(siteName)
+                                                  .append("/")
+                                                  .append(ATTACHMENT_BANNER_TYPE)
+                                                  .append("?r=")
+                                                  .append(token)
```

Comment:
do we need Token generation for Anonymous Access, from Email notifications by example?

```suggestion
```